pywhy_graphs/algorithms/pag.py

```diff
@@ -934,15 +934,23 @@ def is_valid_PAG(graph: PAG) -> bool:
 
     to_remove = []
     to_reorient = []
+    to_replace = []
     for u, v in cedges:
         if (v, u) in dedges:
             to_remove.append((u, v))
         elif (v, u) not in cedges:
             to_reorient.append((u, v))
+        elif (v, u) in cedges and (v, u) not in to_replace:
```

Comment:
The `not in to_replace` does not scale well. You should use a different data structure to check inclusion (e.g. set, or dict)

Reply:
Makes sense.